scripts/get-external-data.py

```diff
@@ -0,0 +1,258 @@
+                        [self._name])
+            if cur.rowcount == 0:
+                cur.execute(('''INSERT INTO "{schema}"."{metadata_table}" '''
+                            '''(name, last_modified) VALUES (%s, %s)''')
+                            .format(schema=self._dst_schema, metadata_table=self._metadata_table),
+                            [self._name, new_last_modified])
+            else:
+                cur.execute('''UPDATE "{schema}"."{metadata_table}" SET last_modified = %s WHERE name = %s'''
+                            .format(schema=self._dst_schema, metadata_table=self._metadata_table),
+                            [new_last_modified, self._name])
+        self._conn.commit()
+
+
+def main():
+    # parse options
+    parser = argparse.ArgumentParser(description="Load external data into a database")
+
+    parser.add_argument("-f", "--force", action="store_true", help="Download new data, even if not required")
+
+    parser.add_argument("-c", "--config", action="store", default="external-data.yml",
+                        help="Name of configuration file (default external-data.yml)")
+    parser.add_argument("-D", "--data", action="store", help="Override data download directory")
+
+    parser.add_argument("-d", "--database", action="store", help="Override database name to connect to")
+    parser.add_argument("-H", "--host", action="store",
+                        help="Override database server host or socket directory")
+    parser.add_argument("-p", "--port", action="store", help="Override database server port")
+    parser.add_argument("-U", "--username", action="store", help="Override database user name")
+    parser.add_argument("-v", "--verbose", action="store_true", help="Be more verbose. Overrides -q")
+    parser.add_argument("-q", "--quiet", action="store_true", help="Only report serious problems")
+
+    opts = parser.parse_args()
+
+    if opts.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    elif opts.quiet:
+        logging.basicConfig(level=logging.WARNING)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    with open(opts.config) as config_file:
+        config = yaml.safe_load(config_file)
+        data_dir = opts.data or config["settings"]["data_dir"]
+        os.makedirs(data_dir, exist_ok=True)
+
+        # If the DB options are unspecified in both on the command line and in the
+        # config file, libpq will pick what to use with the None
+        database = opts.database or config["settings"].get("database")
+        host = opts.host or config["settings"].get("host")
+        port = opts.port or config["settings"].get("port")
+        user = opts.username or config["settings"].get("username")
+        with requests.Session() as s, \
+            psycopg2.connect(database=database,
+                             host=host, port=port,
+                             user=user) as conn:
+
+            s.headers.update({'User-Agent': 'get-external-data.py/osm-carto'})
```

Comment:
What about putting a version number in aswell?

Reply:
Version of the style? then we'd have to somehow automatically update it since we don't want have to manually do that.